daemon/daemon.go

```diff
@@ -91,6 +91,15 @@ func NewDaemon(cfg *config.Config) *Daemon {
 		return nil
 	}
 
+	if cfg.StorageDriver != "" {
+		ctrd.SetSnapshotterName(cfg.StorageDriver)
+	}
```

Comment:
`SetSnapshotterName` should be after check storage driver is available

Reply:
If storage driver is nil, ctrd.CurrentSnapshotterName() will get default snapshotter driver, and default snapshotter also should be checked, in case of other snapshotter has been set before. Before snapshotter check, I try to get containerd version to make sure containerd has booted. @fuweid @Ace-Tang